news/3905.bugfix

```diff
@@ -0,0 +1,8 @@
+Changed rules on where pip version lock file selfcheck.json is stored.
```

Comment:
While I applaud you for writing such an extensive news file entry, I think that it is probably overly descriptive compared to the level of impact the typical user is going to see. For most people this will be entirely transparent. I would suggest trimming this down to 1, maybe two sentences. For a look at our existing change log entries, check out https://github.com/pypa/pip/blob/master/NEWS.rst.

Reply:
There's also a note about this in https://github.com/pypa/pip/blob/master/docs/development.rst#adding-a-news-entry

> In order to maintain a consistent style in the ``NEWS.rst`` file, it is preferred to keep the news entry to the point, in sentence case, shorter than 80 characters and in an imperative tone -- an entry should complete the sentence "This change will ...". In rare cases, where one line is not enough, use a summary line in an imperative tone followed by a blank line separating it from a description of the feature/change in one or more paragraphs, each wrapped at 80 characters. Remember that a news entry is meant for end users and should only contain details relevant to an end user.